### PR TITLE
Allow @Main to have arguments

### PR DESCRIPTION
**Pull request by xtrm-en**
hi

### CODE REVIEW

**Review by pauliesnug — DISMISSED** [review]
hug

**Comment by nxtdaydelivery** [comment]
can you merge this into #151 ?

File: versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/PlatformCommandManagerImpl.java [hunk 1]
```diff
@@ -154,23 +152,55 @@ private CommandManager.Pair<String[], CommandManager.InternalCommand> getCommand
                     // work backwards to find the first match
                     CommandManager.InternalCommand command = get(root, argsIn);
                     if (command != null) {
+                        String primaryPath = command.getPrimaryPath()
+                                .replace(DELIMITER + MAIN_METHOD_NAME, "")
+                                .replace(MAIN_METHOD_NAME, "");
+                        int skipArgs = 0;
+                        if (!primaryPath.isEmpty()) skipArgs++;
+                        for (char c : primaryPath.toCharArray()) {
```
[the remark this answers is not in the document]
Comment (xtrm-en) [reply versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/PlatformCommandManagerImpl.java:160]: ....perhaps.... did i forget a builtin string method here lmao

File: versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/PlatformCommandManagerImpl.java [hunk 2]
```diff
@@ -154,23 +152,55 @@ private CommandManager.Pair<String[], CommandManager.InternalCommand> getCommand
                     // work backwards to find the first match
                     CommandManager.InternalCommand command = get(root, argsIn);
                     if (command != null) {
+                        String primaryPath = command.getPrimaryPath()
```
Comment (nxtdaydelivery) [inline versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/PlatformCommandManagerImpl.java:155]: are you sure this doesn't abuse primary paths? have you checked if aliases still work?

Comment (xtrm-en) [reply versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/PlatformCommandManagerImpl.java:155]: This works with `/oneconfig profiles <overloaded main arg>`, `/oneconfig profile <overloaded arg>`, `/ocfg profile <overloaded arg>`, so :+1:

**Review by Wyvest — APPROVED** [review]
Will need to improve later, but for now this is ok

**Pull request merged**